auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (BuildConfig.DEBUG) {
+            Log.d(TAG, "onActivityResult:" + requestCode + ":" + resultCode + ":" + data);
+        }
+
+        switch (requestCode) {
+            case RC_CREDENTIALS_READ:
+                if (resultCode == RESULT_OK) {
+                    // credential selected from SmartLock, log in with that credential
+                    Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
+                    handleCredential(credential);
+
+                    String email = getEmailFromCredential();
+                    String password = getPasswordFromCredential();
+                    if (email != null) {
+                        if (password == null || password.isEmpty()) {
+                            // identifier/provider combination
+                            redirectToIdpSignIn(email, getAccountTypeFromCredential());
+                        } else {
+                            // email/password combination
+                            signInWithEmailAndPassword(email, password);
+                        }
+                    }
+                } else {
+                    // Smart lock selector cancelled, go to the AuthMethodPicker screen
+                    startAuthMethodChoice();
+                }
+                break;
+            case RC_IDP_SIGNIN:
+            case RC_AUTH_METHOD_PICKER:
+            case RC_EMAIL_FLOW:
+                finish(resultCode, data);
+                break;
+            case RC_PLAY_SERVICES:
+                if (resultCode != RESULT_OK) {
+                    finish(resultCode, data);
+                }
+                break;
+        }
+    }
+
+    private String getEmailFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getId();
+    }
+
+    private String getAccountTypeFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getAccountType();
+    }
+
+    private String getPasswordFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getPassword();
+    }
+
+    private void handleCredential(Credential credential) {
+        mCredential = credential;
+
+        if (IdentityProviders.GOOGLE.equals(credential.getAccountType())) {
+            // Google account, rebuild GoogleApiClient to set account name and then try
+            initGoogleApiClient(credential.getId());
+            // Try silent sign-in with Google Sign In API
+            Auth.GoogleSignInApi.silentSignIn(mGoogleApiClient);
+        } else {
+            // Email/password account
+            String status = String.format("Signed in as %s", credential.getId());
+            Log.d(TAG, status);
```

Comment:
nit: We should avoid logging PII (the email).  Let's just say "Signed in with email"